### PR TITLE
Enabling workflows for managed events

### LINKED ISSUE #8288: Managed Event-types not working with workflows
### Issue Summary

Managed event-types can't be added to workflows by either team owner or the user itself.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

- Browser version, screen recording, console logs, network requests: You can make a recording with [Bird Eats Bug](https://birdeatsbug.com/).
- Node.js version
- Anything else that you think could be an issue.

### LINKED ISSUE #8866: [CAL-1661] wrong slug in managed event-types
should be cal.com/username/ 

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1661](https://linear.app/calcom/issue/CAL-1661/wrong-slug-in-managed-event-types)</sub>

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Enabling workflows for managed event types

**Environment**: Staging(main branch) / Production

Fixes: #8866 

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

1. Create managed event type
2. Go to edit created managed event type
3. Go to Workflows tab 
4. Create a Workflow assigning the managed event type

### CODE REVIEW

**Review by CarinaWolli — COMMENTED** [review]
As a member, I can set a managed event type active on my personal workflow. But in the event types settings, my (inactive) personal workflows are not shown. Not sure if we want to allow it or not but it should be consistent

File: yarn.lock [hunk 1]
```diff
@@ -3980,39 +3961,6 @@ __metadata:
   languageName: unknown
   linkType: soft
 
-"@calcom/auth@workspace:apps/auth":
```
[the remark this answers is not in the document]
Comment (leog) [reply yarn.lock]: @hariombalhara you mean running `./git-setup.sh auth` and then `yarn`?

[the remark this answers is not in the document]
Comment (leog) [reply yarn.lock]: Reverted with main

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
When disabling an even type in a team workflow, all event type get disabled after saving: 
https://www.loom.com/share/a80aaf475e2e473a9d9c727c1ccbf405

One other small thing you can see in the loom: 
It shows the wrong number of active event types when disabling or enabling a managed event type in the dropdown

File: packages/features/ee/workflows/components/EventWorkflowsTab.tsx [hunk 2]
```diff
@@ -40,6 +41,8 @@ const WorkflowListItem = (props: ItemProps) => {
   const isActive = activeEventTypeIds.includes(eventType.id);
   const utils = trpc.useContext();
 
+  console.log(workflow.readOnly, props.isChildrenManagedEventType);
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/EventWorkflowsTab.tsx:44]:
```suggestion
```

**Comment by leog** [comment]
> One other small thing you can see in the loom: It shows the wrong number of active event types when disabling or enabling a managed event type in the dropdown

Left it to just reflect selected event type, no children taken into account

**Comment by CarinaWolli** [comment]
the managed event type of an admin user never shows as checked in the dropdown of a personal workflow, even when it is set as active: 

https://www.loom.com/share/18f57610d3684677b9b1bda516beeba3

**Comment by leog** [comment]
> the managed event type of an admin user never shows as checked in the dropdown of a personal workflow, even when it is set as active:

Thanks @CarinaWolli for bringing this to my attention. It should be fixed now. PTAL.

File: apps/web/pages/event-types/index.tsx [hunk 3]
```diff
@@ -490,18 +490,20 @@ export const EventTypeList = ({ group, groupIndex, readOnly, types }: EventTypeL
...
-                                      type="button"
-                                      StartIcon={Code}
-                                      className="w-full rounded-none"
-                                      embedUrl={encodeURIComponent(embedLink)}>
-                                      {t("embed")}
-                                    </EmbedButton>
-                                  </DropdownMenuItem>
                                 </>
                               )}
+                              {!isManagedEventType && isChildrenManagedEventType && (
```
Comment (CarinaWolli) [inline apps/web/pages/event-types/index.tsx:495]: we are not showing the embed button anymore in the dropdown for normal event types anymore

File: packages/trpc/server/routers/viewer/workflows/activateEventType.handler.ts [hunk 4]
```diff
@@ -103,10 +110,12 @@ export const activateEventTypeHandler = async ({ ctx, input }: ActivateEventType
       const isSmsReminderNumberRequired = eventTypeWorkflow.steps.some((step) => {
         return step.action === WorkflowActions.SMS_ATTENDEE && step.numberRequired;
       });
-      await upsertSmsReminderFieldForBooking({
-        workflowId,
-        isSmsReminderNumberRequired,
-        eventTypeId,
+      [eventTypeId].concat(userEventType.children.map((ch) => ch.id)).map(async (evTyId) => {
+        await upsertSmsReminderFieldForBooking({
```
Comment (leog) [inline packages/trpc/server/routers/viewer/workflows/activateEventType.handler.ts:114]: @CarinaWolli does this look good?

Comment (CarinaWolli) [reply packages/trpc/server/routers/viewer/workflows/activateEventType.handler.ts:114]: Yes, that looks good and is actually a good point! We also need to do that when deactivating a workflow, in line 91 `removeSmsReminderFieldForBooking`

And also when activating/deactivating the MET in the workflow itself

File: apps/web/components/eventtype/EventSetupTab.tsx [hunk 5]
```diff
@@ -390,7 +390,12 @@ export const EventSetupTab = (
           addOnLeading={
             <>
               {CAL_URL?.replace(/^(https?:|)\/\//, "")}/
-              {team ? "team/" + team.slug : eventType.users[0].username}/
+              {!isManagedEventType
+                ? team
+                  ? "team/" + team.slug
+                  : eventType.users[0].username
+                : t("username_placeholder")}
+              /
```
Comment (leog) [inline apps/web/components/eventtype/EventSetupTab.tsx:398]: Fixed #8866

File: apps/web/pages/auth/setup/index.tsx [hunk 6]
```diff
@@ -172,10 +175,9 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
         agreedLicenseAt: new Date(),
       },
     });
-    deploymentKey = await getDeploymentKey(prisma);
   }
 
-  const isFreeLicense = deploymentKey === "";
+  const isFreeLicense = (await getDeploymentKey(prisma)) === "";
```
Comment (leog) [inline apps/web/pages/auth/setup/index.tsx:180]: These changes are needed to fix checking correctly existing environment variable for a license key in order to create the proper record in the database automatically.

File: apps/web/pages/event-types/index.tsx (hunk 3 above)
Comment (leog) [reply apps/web/pages/event-types/index.tsx]: Fixed!

File: packages/trpc/server/routers/viewer/workflows/activateEventType.handler.ts [hunk 7]
```diff
@@ -118,13 +121,15 @@ export const activateEventTypeHandler = async ({ ctx, input }: ActivateEventType
...
-        eventTypeId,
+        eventTypeId: { in: [eventTypeId].concat(userEventType.children.map((ch) => ch.id)) },
       },
     });
 
-    await removeSmsReminderFieldForBooking({
-      workflowId,
-      eventTypeId,
+    [eventTypeId].concat(userEventType.children.map((ch) => ch.id)).map(async (chId) => {
+      await removeSmsReminderFieldForBooking({
```
Comment (leog) [inline packages/trpc/server/routers/viewer/workflows/activateEventType.handler.ts:129]: Done here too now @CarinaWolli

File: packages/trpc/server/routers/viewer/workflows/update.handler.ts [hunk 8]
```diff
@@ -666,7 +714,7 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
     });
   }
 
-  for (const eventTypeId of activeOn) {
+  for (const eventTypeId of activeOnWithChildren) {
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/update.handler.ts:717]: @leog I just pushed a fix for showing the SMS reminder field correctly. `activeOnWithChildren` should now include also children event types and `removedEventTypes` also has the children event types included now

**Review by CarinaWolli — APPROVED** [review]
Looks good to me. Let's not auto-merge for now. I've made some final changes that need to be reviewed by @leog 🙏

File: packages/trpc/server/routers/viewer/workflows/update.handler.ts (hunk 8 above)
Comment (leog) [reply packages/trpc/server/routers/viewer/workflows/update.handler.ts:717]: Looks good, queuing the merge, thanks @CarinaWolli!

**Pull request merged**